### ISSUE TITLE
Godot 3.2.1: 'files_dropped' signal not emitting

### ISSUE THREAD

**Issue by grotar00** [godotengine/godot#37315]
**Godot version:** 3.2.1.stable.official

**OS/device including version:** Windows 10 x64

**Issue description:** Engine doesn't react when a file or multiple files are dropped over game window. Sample project still works with 3.2 build.

**Steps to reproduce:** connect _SceneTree_ with _files_dropped_ signal.

**Minimal reproduction project:** [FileDropTest.zip](https://github.com/godotengine/godot/files/4386951/FileDropTest.zip)

**Comment by KoBeWi** [comment]
The project works for me correctly in 3.2.1.

**Comment by grotar00** [comment]
Solved by unchecking this box in exucutable properties. Launching engine as administrator from context menu yields same effect.
Problem appears in older versions as well up to Godot 2.1.4 so issue's title is misleading.
![image](https://user-images.githubusercontent.com/8709735/77689862-77535300-6fb3-11ea-8420-461fead3d145.png)

**Comment by KoBeWi** [comment]
I ran the editor as administrator and tried to drop the test project into the manager and wondered why it doesn't work .-. So yeah, confirmed.

**Comment by akien-mga** [comment]
I guess that might be a security feature of Windows? But if so there should be some kind of error we can intercept and pass on to the user.

**Comment by Anutrix** [comment]
One thing I found out is that this is a Windows behaviour. Low integrity processes (explorer.exe runs with low integrity) are not allowed for drag n drop with higher integrity processes (godot).

**Comment by akien-mga** [comment]
> One thing I found out is that this is Windows behaviour. Low integrity processes (explorer.exe runs with low integrity) are not allowed for drag n drop with higher integrity processes (Godot).
> Unfortunately, there are only unstable undocumented ways to allow this. There is also no recent data about this either.
> I did find [social.msdn.microsoft.com/Forums/windowsdesktop/en-US/0ccf84fd-b78d-45b3-9b79-7366003cb19d/wmdropfiles-in-an-elevated-application-administrator?forum=windowsgeneraldevelopmentissues](https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/0ccf84fd-b78d-45b3-9b79-7366003cb19d/wmdropfiles-in-an-elevated-application-administrator?forum=windowsgeneraldevelopmentissues) though.

Indeed, also found the same workaround documented here: https://helgeklein.com/blog/2010/03/how-to-enable-drag-and-drop-for-an-elevated-mfc-application-on-vistawindows-7/

We could try that but I'm not sure it's worth the trouble, and I don't know how these commands impact drag and drop when not using Godot in elevated mode (which is the normal, supported use case).

So another option is to close this as something we don't want to attempt to workaround, since it seems to go against Windows' design.

WDYT @bruvzg @vnen @hpvb?

**Comment by Calinou** [comment]
Can we detect from Godot if the editor is being run as Administrator? I'd warn the user about this on startup.

**Comment by bruvzg** [comment]
> Unfortunately, there are only unstable undocumented ways to allow this. There is also no recent data about this either.
I did find https://social.msdn.microsoft.com/Forums/windowsdesktop/en-US/0ccf84fd-b78d-45b3-9b79-7366003cb19d/wmdropfiles-in-an-elevated-application-administrator?forum=windowsgeneraldevelopmentissues though.

`ChangeWindowMessageFilterEx` variant is [documented](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-changewindowmessagefilterex) and supported.

```c++
ChangeWindowMessageFilterEx(wd.hWnd, WM_DROPFILES, MSGFLT_ALLOW, nullptr);
ChangeWindowMessageFilterEx(wd.hWnd, WM_COPYDATA, MSGFLT_ALLOW, nullptr);
ChangeWindowMessageFilterEx(wd.hWnd, WM_COPYGLOBALDATA, MSGFLT_ALLOW, nullptr);
```
